Add d-block to bootstrap 4 alerts

normisg opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26869
| License       | MIT
| Doc PR        | N/A

nicolas-grekas committed:
[DI] Improve error message for non-autowirable scalar argument

nicolas-grekas committed:
[HttpKernel] Dont create mock cookie for new sessions in tests

Commit:
Add d-block to bootstrap 4 alerts

Tobion committed:
minor #26856 [DI] Improve error message for non-autowirable scalar argument (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

[DI] Improve error message for non-autowirable scalar argument

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26838
| License       | MIT
| Doc PR        | -

Commits
-------

7f398117f4 [DI] Improve error message for non-autowirable scalar argument

chapterjason commented:
Is there any reason why a `span` is used to display the alerts? When I look at the docs of bootstrap, I can clearly see that the `div` tag are used for that.

https://getbootstrap.com/docs/4.0/components/alerts/

Using `d-block` is for me really odd.

normisg commented:
Yes, it was changed in https://github.com/symfony/symfony/commit/5dcaaadb5e8c01148b67d009c95a5118849f9a55 to "Make sure form errors is valid HTML".
Seems unnecessary to me too.

vudaltsov commented:
The reason for this is that now error blocks might appear inside `<label>` tags. You cannot have `<div>` inside a label.

vudaltsov commented on src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig:
```diff
@@ -283,7 +283,7 @@
 {% block form_errors -%}
     {%- if errors|length > 0 -%}
-        <span class="{% if form is not rootform %}invalid-feedback d-block{% else %}alert alert-danger{% endif %}">
+        <span class="{% if form is not rootform %}invalid-feedback d-block{% else %}alert alert-danger d-block{% endif %}">
```
Let's move the `d-block` class outside the `{% if %}` statement to have it only once in the line

nicolas-grekas committed:
bug #26857 [HttpKernel] Dont create mock cookie for new sessions in tests (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

[HttpKernel] Dont create mock cookie for new sessions in tests

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26493
| License       | MIT
| Doc PR        | -

@romainneutron can you please check if this fixes your issue?

Commits
-------

86b850e [HttpKernel] Dont create mock cookie for new sessions in tests

nicolas-grekas commented:
Thank you @normisg.
